modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -83,115 +82,163 @@ def _read(cls, filepath_or_buffer, **kwargs):
-            parse_dates=parse_dates,
+            skiprows=1 if encoding is not None else None,
             usecols=usecols,
+            nrows=None,
+            compression=compression_infered,
+            index_col=index_col,
         )
-        encoding = kwargs.get("encoding", None)
-        quotechar = kwargs.get("quotechar", '"').encode(
-            encoding if encoding is not None else "UTF-8"
-        )
-        is_quoting = kwargs.get("quoting", "") != csv.QUOTE_NONE
-        with cls.file_open(filepath_or_buffer, "rb", compression_type) as f:
-            # Skip the header since we already have the header information and skip the
-            # rows we are told to skip.
-            if isinstance(skiprows, int) or skiprows is None:
-                if skiprows is None:
-                    skiprows = 0
-                header = kwargs.get("header", "infer")
-                if header == "infer" and kwargs.get("names", None) is None:
-                    skiprows += 1
-                elif isinstance(header, int):
-                    skiprows += header + 1
-                elif hasattr(header, "__iter__") and not isinstance(header, str):
-                    skiprows += max(header) + 1
-            if kwargs.get("encoding", None) is not None:
-                partition_kwargs["skiprows"] = 1
-            # Launch tasks to read partitions
-            partition_ids = []
-            index_ids = []
-            dtypes_ids = []
-            # Max number of partitions available
-            num_partitions = NPartitions.get()
-            # This is the number of splits for the columns
-            num_splits = min(len(column_names), num_partitions)
-            # Metadata
-            column_chunksize = compute_chunksize(empty_pd_df, num_splits, axis=1)
-            if column_chunksize > len(column_names):
-                column_widths = [len(column_names)]
-                # This prevents us from unnecessarily serializing a bunch of empty
-                # objects.
-                num_splits = 1
-            else:
-                column_widths = [
-                    column_chunksize
-                    if len(column_names) > (column_chunksize * (i + 1))
-                    else 0
-                    if len(column_names) < (column_chunksize * i)
-                    else len(column_names) - (column_chunksize * i)
-                    for i in range(num_splits)
-                ]
-
-            args = {
-                "fname": filepath_or_buffer,
-                "num_splits": num_splits,
-                **partition_kwargs,
-            }
+
+        with cls.file_open(filepath_or_buffer_md, "rb", compression_infered) as f:
+            rows_skipper = cls.rows_skipper_builder(f, quotechar, is_quoting=is_quoting)
+            rows_skipper(skiprows)
 
             splits = cls.partitioned_file(
                 f,
                 num_partitions=num_partitions,
-                nrows=nrows,
-                skiprows=skiprows,
```

Comment:
why not just skip rows inside `partitioned_file` func?

Reply:
This is done to make `partitioned_file` function lighter. The plan is to handle trivial integer `skiprows` in the `CSVDispatcher._read` function and complicated `skiprows` cases (callables or arrays) in the `TextFileDispatcher.partitioned_file` function.

Reply:
Why solve one problem in two different places?

Reply:
The reason is that when `skiprows` will be non-integer `partitioned_file` will become complicated, but, i think, you are right - handling of `skiprows` should be done in one place, will fix it soon.